src/timeline-list/index.tsx

```diff
@@ -60,21 +60,24 @@ const TimelineList = (props: TimelineListProps) => {
   const {pages, pagesRef, resetPages, resetPagesDebounce, scrollToPageDebounce, shouldResetPages, isOutOfRange} =
     useTimelinePages({date, listRef, numberOfDays});
 
-  useEffect(() => {
-    if (date !== prevDate.current) {
-      const datePageIndex = pagesRef.current.indexOf(date);
-
-      if (updateSource !== UpdateSources.LIST_DRAG) {
-        if (isOutOfRange(datePageIndex)) {
-          updateSource === UpdateSources.DAY_PRESS ? resetPages(date) : resetPagesDebounce(date);
-        } else {
-          scrollToPageDebounce(datePageIndex);
-        }
+  const scrollToCurrentDate = useCallback((date: string) => {
+    const datePageIndex = pagesRef.current.indexOf(date);
+
+    if (updateSource !== UpdateSources.LIST_DRAG) {
+      if (isOutOfRange(datePageIndex)) {
+        updateSource === UpdateSources.DAY_PRESS ? resetPages(date) : resetPagesDebounce(date);
+      } else {
+        scrollToPageDebounce(datePageIndex);
       }
+    }
+    prevDate.current = date;
+  }, [pagesRef.current, updateSource]);
```

Comment:
pagesRef.current is not dep...

Reply:
why not? it changes when the number of days changes (since when numOfDay = undefined we have a page for each day and when the number of days = 3 we have a page for every 3 days), when the scroll doesn't adjust to it we get an issue with the scrolling (it thinks that it scrolled 3 pages instead of 1 because of the pagesRef old value being used

Reply:
`Mutable values like 'something.current' aren't valid dependencies because mutating them doesn't re-render the component.`

Reply:
removed unnecessary dependency